upgrade/v102to110/upgrade.go

```diff
@@ -113,17 +113,16 @@ func upgradeVolumes(namespace string, lhClient *lhclientset.Clientset) (err erro
 		// in pr https://github.com/longhorn/longhorn-manager/pull/789
 		// we added a new access mode field, that is exposed to the ui
 		// so we add the previously only supported rwo access mode
+		requireUpdate := false
 		if v.Spec.AccessMode == "" {
 			v.Spec.AccessMode = types.AccessModeReadWriteOnce
-			updatedVolume, err := lhClient.LonghornV1beta1().Volumes(namespace).Update(context.TODO(), &v, metav1.UpdateOptions{})
-			if err != nil {
-				return err
-			}
-			v = *updatedVolume
+			requireUpdate = true
 		}
-
-		if v.Status.Robustness == types.VolumeRobustnessDegraded {
+		if v.Status.Robustness == types.VolumeRobustnessDegraded && v.Status.LastDegradedAt == "" {
 			v.Status.LastDegradedAt = util.Now()
+			requireUpdate = true
+		}
+		if requireUpdate {
 			if _, err := lhClient.LonghornV1beta1().Volumes(namespace).UpdateStatus(context.TODO(), &v, metav1.UpdateOptions{}); err != nil {
```

Comment:
When `requireUpdate` is true, you only update the volume status but actually, but how about the change of `v.spec.accessmode` which will not be taken effect if only updating the status.